src/connectionbenchmark/connectionbenchmark.cpp

```diff
@@ -75,7 +76,7 @@ void ConnectionBenchmark::start() {
   setState(StateRunning);
 
   if (vpn->connectionHealth()->stability() == ConnectionHealth::NoSignal) {
-    handleStabilityChange();
+    TimerSingleShot::create(this, 3000, [this]() { handleStabilityChange(); });
```

Comment:
This sounds like a hack. The "real" fix should be to monitor the connection stability and at the next good connection state, we restart the benckmark.